sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ResearchStackUploadArchiveFactory.kt

```diff
@@ -0,0 +1,172 @@
+/*
+ * BSD 3-Clause License
+ *
+ * Copyright 2018  Sage Bionetworks. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1.  Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * 2.  Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation and/or
+ * other materials provided with the distribution.
+ *
+ * 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission. No license is granted to the trademarks of
+ * the copyright holders even if such marks are included in this software.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.sagebionetworks.research.sageresearch.viewmodel
+
+import android.os.Build
+import com.google.common.collect.ImmutableList
+import org.joda.time.DateTime
+import org.researchstack.backbone.result.Result
+import org.sagebionetworks.bridge.android.BridgeConfig
+import org.sagebionetworks.bridge.android.manager.upload.ArchiveUtil
+import org.sagebionetworks.bridge.data.Archive
+import org.sagebionetworks.bridge.data.JsonArchiveFile
+import org.sagebionetworks.bridge.researchstack.TaskHelper
+import org.sagebionetworks.bridge.researchstack.factory.ArchiveFileFactory
+import org.sagebionetworks.bridge.rest.RestUtils
+import org.sagebionetworks.research.sageresearch.dao.room.ScheduledActivityEntity
+import org.sagebionetworks.research.sageresearch.dao.room.bridgeMetadataCopy
+import org.slf4j.LoggerFactory
+import java.util.UUID
+
+/**
+ * The ResearchStackUploadArchiveFactory controls upload archive creation and allows for
+ * easy sub-classing to change the functionality of the factory.
+ */
+open class ResearchStackUploadArchiveFactory: ArchiveFileFactory() {
```

Comment:
This class should be in `researchstack-sdk`. Not sure how a `researchstack-sdk` dependency got added to `sageresearch-app-sdk` but we should be keeping them separate. Can you add a TODO?